public/app/features/dashboard-scene/scene/DashboardMacro.ts

```diff
@@ -24,7 +24,8 @@ class DashboardMacro implements FormatVariable {
       case 'name':
       case 'id':
       default:
-        return dashboard.state.title;
+        // URL encode the dashboard title when it's used in URLs
+        return fieldPath === 'raw' ? dashboard.state.title : encodeURIComponent(dashboard.state.title);
```

Comment:
I don't think this is fix the issue. It is not a matter of encoding.
We need to lowercase the title and then replace spaces ` ` with `-`.
At least this is what I doing as a workaround.